Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2RobotImporterEditorSystemComponent.h"
+#include "RobotImporterWidget.h"
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzToolsFramework/API/ViewPaneOptions.h>
+
+namespace ROS2
+{
+    void ROS2RobotImporterEditorSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<ROS2RobotImporterEditorSystemComponent, ROS2RobotImporterSystemComponent>()->Version(0);
+        }
+    }
+
+    void ROS2RobotImporterEditorSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        ROS2RobotImporterSystemComponent::GetProvidedServices(provided);
+        provided.push_back(AZ_CRC_CE("ROS2RobotImporterEditorService"));
+    }
+
+    void ROS2RobotImporterEditorSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        ROS2RobotImporterSystemComponent::GetIncompatibleServices(incompatible);
+        incompatible.push_back(AZ_CRC_CE("ROS2RobotImporterEditorService"));
+    }
+
+    void ROS2RobotImporterEditorSystemComponent::Activate()
+    {
+        ROS2RobotImporterSystemComponent::Activate();
+        AzToolsFramework::EditorEvents::Bus::Handler::BusConnect();
+    }
+
+    void ROS2RobotImporterEditorSystemComponent::Deactivate()
+    {
+        AzToolsFramework::EditorEvents::Bus::Handler::BusDisconnect();
+        ROS2RobotImporterSystemComponent::Deactivate();
+    }
+
+    void ROS2RobotImporterEditorSystemComponent::NotifyRegisterViews()
+    {
+        AzToolsFramework::ViewPaneOptions options;
+        options.paneRect = QRect(100, 100, 500, 500); // TODO - config/UX sensitive
+        options.showOnToolsToolbar = true;
+        // options.toolbarIcon = ":/.svg"; // TODO - icon
+
+        AzToolsFramework::RegisterViewPane<RobotImporterWidget>("Robot Importer", "ROS2", options);
+    }
+
+    ROS2RobotImporterEditorSystemComponent::ROS2RobotImporterEditorSystemComponent() = default;
+
+    ROS2RobotImporterEditorSystemComponent::~ROS2RobotImporterEditorSystemComponent() = default;
+} // namespace ROS2
```

Comment:
minor:
You could also declare them as default in the header file instead of the cpp file

Reply:
see above